test/format/config.c

```diff
@@ -251,10 +251,11 @@ config_table(TABLE *table, void *arg)
     table->max_mem_page = MEGABYTE(TV(BTREE_MEMORY_PAGE_MAX));
 
     /*
-     * Keep the number of rows and keys/values small for in-memory runs (overflow items aren't an
-     * issue for in-memory configurations and it helps prevents cache overflow).
+     * Keep the number of rows and keys/values small for in-memory and direct I/O runs (overflow
+     * items aren't an issue for in-memory configurations and it helps prevents cache overflow, and
+     * direct I/O can be so slow the additional I/O for overflow items causes eviction to stall).
      */
-    if (GV(RUNS_IN_MEMORY)) {
+    if (GV(RUNS_IN_MEMORY) || GV(DISK_DIRECT_IO)) {
```

Comment:
Do we need to tweak `internal_page_max` and `leaf_page_max` too to make sure there are no overflow items?

Reply:
@keithbostic, should we set those values too? if I understand correctly, do we want to set those values to a reasonably big number to be able to contain the overflow items?

Reply:
> @keithbostic, should we set those values too? if I understand correctly, do we want to set those values to a reasonably big number to be able to contain the overflow items?

@etienneptl, I don't think you'll need to do that -- the settings for key/value max are going to prevent overflow items in general.